Content.Server/Antag/MobReplacementRuleSystem.cs

```diff
@@ -1,38 +1,179 @@
+    [Dependency] private readonly AdvertiseSystem _advertise = default!;
+
 
     protected override void Started(EntityUid uid, MobReplacementRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
         base.Started(uid, component, gameRule, args);
 
         var query = AllEntityQuery<VendingMachineComponent, TransformComponent>();
         var spawns = new List<(EntityUid Entity, EntityCoordinates Coordinates)>();
+        var stations = _gameTicker.GetSpawnableStations();
 
         while (query.MoveNext(out var vendingUid, out _, out var xform))
         {
-            if (!_random.Prob(component.Chance))
+            var ownerStation = _station.GetOwningStation(vendingUid);
+
+            if (ownerStation == null
+                || ownerStation != stations[0])
+                continue;
+
+            // Make sure that we aren't running this on something that is already a mimic
+            if (HasComp<CombatModeComponent>(vendingUid))
                 continue;
 
             spawns.Add((vendingUid, xform.Coordinates));
         }
 
-        foreach (var entity in spawns)
+        if (spawns == null)
+        {
+            //WTF THE STATION DOESN'T EXIST! WE MUST BE IN A TEST! QUICK, PUT A MIMIC AT 0,0!!!
+            Spawn(component.Proto, new EntityCoordinates(uid, new Vector2(0, 0)));
+        }
+        else
         {
-            var coordinates = entity.Coordinates;
-            Del(entity.Entity);
+            // This is intentionally not clamped. If a server host wants to replace every vending machine in the entire station with a mimic, who am I to stop them?
+            var k = MathF.MaxMagnitude(component.NumberToReplace, 1);
+            while (k > 0 && spawns != null && spawns.Count > 0)
+            {
+                if (k > 1)
+                {
+                    var spawnLocation = _random.PickAndTake(spawns);
+                    BuildAMimicWorkshop(spawnLocation.Entity, component);
+                }
+                else
+                {
+                    BuildAMimicWorkshop(spawns[0].Entity, component);
+                }
+
+                if (k == MathF.MaxMagnitude(component.NumberToReplace, 1)
+                    && component.DoAnnouncement)
+                    _chat.DispatchStationAnnouncement(stations[0], Loc.GetString("station-event-rampant-intelligence-announcement"), playDefaultSound: true,
+                        colorOverride: Color.Red, sender: "Central Command");
+
+                k--;
+            }
+        }
+    }
+
+    /// <summary>
+    /// 	It's like Build a Bear, but MURDER
+    /// </summary>
+    /// <param name="uid"></param>
+    public void BuildAMimicWorkshop(EntityUid uid, MobReplacementRuleComponent component)
+    {
+        var metaData = MetaData(uid);
+        var vendorPrototype = metaData.EntityPrototype;
+        var mimicProto = _prototype.Index<EntityPrototype>(component.Proto);
```

Comment:
Since `component.Proto` is already EntProtoId type specification can be removed.
```suggestion
        var mimicProto = _prototype.Index(component.Proto);
```